frame/dex-router/src/lib.rs

```diff
@@ -109,52 +105,66 @@ pub mod pallet {
 			who: T::AccountId,
 			x_asset_id: T::AssetId,
 			y_asset_id: T::AssetId,
-			route: Vec<DexRouteNode<T::PoolId>>,
+			route: Vec<T::PoolId>,
 		},
 		RouteDeleted {
 			who: T::AccountId,
 			x_asset_id: T::AssetId,
 			y_asset_id: T::AssetId,
-			route: Vec<DexRouteNode<T::PoolId>>,
+			route: Vec<T::PoolId>,
 		},
 		RouteUpdated {
 			who: T::AccountId,
 			x_asset_id: T::AssetId,
 			y_asset_id: T::AssetId,
-			old_route: Vec<DexRouteNode<T::PoolId>>,
-			updated_route: Vec<DexRouteNode<T::PoolId>>,
+			old_route: Vec<T::PoolId>,
+			updated_route: Vec<T::PoolId>,
 		},
 	}
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {}
 
 	impl<T: Config> Pallet<T> {
+		fn check_route(
```

Comment:
Can we have tests checking the failure case?

Reply:
update_route() test covers all possible errors.
https://github.com/ComposableFi/composable/pull/899/files#diff-183beb853fd7716a87bad38cf794b5a91acdf4cf98dd86d52e2a499760ad5abfR148